yjit_codegen.c

```diff
@@ -717,67 +722,57 @@ enum {
     OSWB_MAX_DEPTH = 5,           // up to 5 different classes
 };
 
+// Codegen for getting an instance variable.
+// Preconditions:
+//   - receiver is in REG0
+//   - receiver has the same class as CLASS_OF(comptime_receiver)
+//   - no stack push or pops to ctx since the entry to the codegen of the instruction being compiled
 static codegen_status_t
-gen_getinstancevariable(jitstate_t* jit, ctx_t* ctx)
+gen_get_ivar(jitstate_t *jit, ctx_t *ctx, const int max_chain_depth, VALUE comptime_receiver, ID ivar_name, bool pop_receiver, uint8_t *side_exit)
```

Comment:
I have a feeling you might not like this suggestion 😅  but we could use `OPND_SELF` / `OPND_STACK` instead of `pop_receiver`, would be more legible than true/false?

Reply:
Tried it in https://github.com/Shopify/ruby/pull/94/commits/40c3088ae155b4f51d60fe68f55e215f4b4b396c.
The two callers read better but the function itself becomes a bit more cryptic, to the point of needing a comment 🤷🏼